torch/distributed/fsdp/sharded_grad_scaler.py

```diff
@@ -183,100 +132,47 @@ def _foreach_non_finite_check_and_unscale_cpu_(
                 torch.isinf(grad).any().item() is True
                 or torch.isnan(grad).any().item() is True
             ):
-                found_inf.data = torch.tensor([1.0])
```

Comment:
fill_(1.0) seems better because it won't change shape from [] to [1]

Reply:
I am all for removing .data calls!